c10/cuda/CUDACachingAllocator.h

```diff
@@ -175,6 +177,14 @@ struct SnapshotInfo {
   std::vector<std::vector<TraceEntry>> device_traces;
 };
 
+// returns the pointers freed in the pool
+// and the pointers allocated. Note: a pointer
+// may appear in both freed and allocated
```

Comment:
So what does it mean if a pointer appears in both freed and allocated? Is it live coming out or not?

Reply:
It could be reallocated as a separate size.

Reply:
Can I have the opposite situation? A tensor is allocated and then freed internally?

Reply:
Yes that is possible. We pass in the stale storages to the api and them swap their deleters with no-ops so they are not freed twice.